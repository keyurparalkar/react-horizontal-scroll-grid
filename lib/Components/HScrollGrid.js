"use strict";

function _typeof(obj) { "@babel/helpers - typeof"; if (typeof Symbol === "function" && typeof Symbol.iterator === "symbol") { _typeof = function _typeof(obj) { return typeof obj; }; } else { _typeof = function _typeof(obj) { return obj && typeof Symbol === "function" && obj.constructor === Symbol && obj !== Symbol.prototype ? "symbol" : typeof obj; }; } return _typeof(obj); }

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;

var _react = _interopRequireDefault(require("react"));

require("../Stylesheets/HScrollGrid.css");

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } }

function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); return Constructor; }

function _inherits(subClass, superClass) { if (typeof superClass !== "function" && superClass !== null) { throw new TypeError("Super expression must either be null or a function"); } subClass.prototype = Object.create(superClass && superClass.prototype, { constructor: { value: subClass, writable: true, configurable: true } }); if (superClass) _setPrototypeOf(subClass, superClass); }

function _setPrototypeOf(o, p) { _setPrototypeOf = Object.setPrototypeOf || function _setPrototypeOf(o, p) { o.__proto__ = p; return o; }; return _setPrototypeOf(o, p); }

function _createSuper(Derived) { var hasNativeReflectConstruct = _isNativeReflectConstruct(); return function _createSuperInternal() { var Super = _getPrototypeOf(Derived), result; if (hasNativeReflectConstruct) { var NewTarget = _getPrototypeOf(this).constructor; result = Reflect.construct(Super, arguments, NewTarget); } else { result = Super.apply(this, arguments); } return _possibleConstructorReturn(this, result); }; }

function _possibleConstructorReturn(self, call) { if (call && (_typeof(call) === "object" || typeof call === "function")) { return call; } return _assertThisInitialized(self); }

function _assertThisInitialized(self) { if (self === void 0) { throw new ReferenceError("this hasn't been initialised - super() hasn't been called"); } return self; }

function _isNativeReflectConstruct() { if (typeof Reflect === "undefined" || !Reflect.construct) return false; if (Reflect.construct.sham) return false; if (typeof Proxy === "function") return true; try { Date.prototype.toString.call(Reflect.construct(Date, [], function () {})); return true; } catch (e) { return false; } }

function _getPrototypeOf(o) { _getPrototypeOf = Object.setPrototypeOf ? Object.getPrototypeOf : function _getPrototypeOf(o) { return o.__proto__ || Object.getPrototypeOf(o); }; return _getPrototypeOf(o); }

var HScrollGrid = /*#__PURE__*/function (_React$Component) {
  _inherits(HScrollGrid, _React$Component);

  var _super = _createSuper(HScrollGrid);

  function HScrollGrid(props) {
    var _this;

    _classCallCheck(this, HScrollGrid);

    _this = _super.call(this, props);
    _this.hscrollRef = /*#__PURE__*/_react.default.createRef();
    return _this;
  }

  _createClass(HScrollGrid, [{
    key: "componentDidMount",
    value: function componentDidMount() {
      var n = this.hscrollRef.current ? this.hscrollRef.current.children.length : 0;
      var gW = "".concat(this.props.gridWidth, "px");
      var gH = "".concat(this.props.gridHeight, "px");
      var cW = "".concat(this.props.cardWidth, "px");
      var cardBgColor = this.props.backgroundColor ? this.props.backgroundColor : 'transparent';

      if (this.hscrollRef.current) {
        this.hscrollRef.current.style.setProperty('--total', n);
        this.hscrollRef.current.style.setProperty('--gridWidth', gW);
        this.hscrollRef.current.style.setProperty('--gridHeight', gH);
        this.hscrollRef.current.style.setProperty('--cardWidth', cW);
        this.hscrollRef.current.style.setProperty('--cBgCol', cardBgColor);
      }
    }
  }, {
    key: "render",
    value: function render() {
      return /*#__PURE__*/_react.default.createElement("ul", {
        className: "hscroll-grid",
        "data-testid": "test-ul",
        ref: this.hscrollRef
      }, this.props.children);
    }
  }]);

  return HScrollGrid;
}(_react.default.Component);

var _default = HScrollGrid;
exports.default = _default;