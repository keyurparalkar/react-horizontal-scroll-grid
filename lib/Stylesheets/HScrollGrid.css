ul.hscroll-grid{
    width:var(--gridWidth);
    height: var(--gridHeight);

    display:grid;
    grid-template-columns: repeat(var(--total), var(--cardWidth)) 10px;
    grid-column-gap: 20px;

    overflow-x:scroll;
    overflow-y:hidden;
}   

li{
    width: var(--cardWidth);
    height: var(--gridHeight);
    background-color: var(--cBgCol);
    display: flex;
    flex-direction: row;
    justify-content: center;
    align-items: center;

    box-shadow: 0 4px 8px 0 rgba(0,0,0,0.1);
    transition: 0.3s;
}

li:hover{
    box-shadow: 0 8px 16px 0 rgba(0,0,0,0.2);
}