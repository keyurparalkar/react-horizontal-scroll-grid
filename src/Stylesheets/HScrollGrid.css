ul.hscroll-grid{
    width:400px;
    height: 100px;

    display:grid;
    grid-template-columns: repeat(var(--total), 100px) 10px;
    grid-column-gap: 10px;

    overflow-x:scroll;
    overflow-y:hidden;
}   

li{
    width: 100px;
    height: 100px;
    background-color: antiquewhite;
    display: flex;
    flex-direction: row;
    justify-content: center;
    align-items: center;
}