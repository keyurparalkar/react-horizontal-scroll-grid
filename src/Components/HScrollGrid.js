import React from 'react';
import '../Stylesheets/HScrollGrid.css';

class HScrollGrid extends React.Component{
    constructor(props){
        super(props);

        this.hscrollRef = React.createRef();
    }

    componentDidMount(){
        let n = this.hscrollRef.current ? this.hscrollRef.current.children.length : 0;
        let gW = `${this.props.gridWidth}px`;
        let gH = `${this.props.gridHeight}px`;
        let cW = `${this.props.cardWidth}px`;
        let cardBgColor = this.props.backgroundColor ? this.props.backgroundColor : 'transparent';

        if(this.hscrollRef.current){
            this.hscrollRef.current.style.setProperty('--total', n);
            this.hscrollRef.current.style.setProperty('--gridWidth', gW);
            this.hscrollRef.current.style.setProperty('--gridHeight', gH);
            this.hscrollRef.current.style.setProperty('--cardWidth', cW);
            this.hscrollRef.current.style.setProperty('--cBgCol', cardBgColor);
        }
    }
    render(){
        return(
            <ul className="hscroll-grid" data-testid="test-ul" ref={this.hscrollRef}>
                {this.props.children}
            </ul>
        );
    }
}

export default HScrollGrid;