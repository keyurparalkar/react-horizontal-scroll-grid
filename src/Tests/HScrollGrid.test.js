/**
 *  @jest-environment jsdom
 */

import React from 'react';
import { shallow, mount, configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import '@testing-library/jest-dom';
import { render, screen } from '@testing-library/react';

import HScrollGrid from '../Components/HScrollGrid.js';


configure({ adapter: new Adapter() });

// it("Matches Snapshots", ()=> {
//     const { asFragment } = render(<HScrollGrid/>);

//     expect(asFragment()).toMatchSnapshot();
// });

it('Does Root Element Exists?', () => {
    // const keys = [1, 2];
    // const cards = keys.map(elem => <li key={elem}> Test </li>);
    const cards = [<li key="1">Test</li>]
    const mockElem0 = shallow(<HScrollGrid>{cards}
    </HScrollGrid>);

    expect(mockElem0.find('.hscroll-grid')).toHaveLength(1);
});

it('Does BackgroundColor exits for children?', () => {
    const cards = [<li key="1">Test</li>]
    const { getByTestId } = render(<HScrollGrid
        gridWidth={400}
        gridHeight={100}
        cardWidth={100}
        backgroundColor="blue">{cards}</HScrollGrid>);

        const ulNode = getByTestId('test-ul');
        // console.log(ulNode.style['_values']['--cBgCol'])

        expect(ulNode.style['_values']['--cBgCol']).toBe("blue");
    // expect(mockElem1.container.firstChild).toHaveClass('hscroll-grid');
    // expect(mockElem1.container.firstChild).toHaveStyle('width:400px');
});
