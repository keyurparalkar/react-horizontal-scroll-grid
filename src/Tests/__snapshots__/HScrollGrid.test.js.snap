// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Matches Snapshots 1`] = `
<DocumentFragment>
  <ul
    class="hscroll-grid"
    data-testid="test-ul"
    style="--total: 0; --gridWidth: undefinedpx; --gridHeight: undefinedpx; --cardWidth: undefinedpx; --cBgCol: transparent;"
  />
</DocumentFragment>
`;
